internal/cache.go

```diff
@@ -20,8 +20,9 @@ import (
 )
 
 // NewStreamData returns a stream data for anything that we need to store in streams.
-func NewStreamData(data []byte) *StreamData {
+func NewStreamData(md []byte, data []byte) *StreamData {
```

Comment:
Metadata is anything that the caller is wanted to tie to the stream message so that StreamData can be generic. For the case of real-time, it is `clientId`, `eventName`, etc - there is a struct for this in serialization.go (`StreamMessageMD`)